runtime/autoload/format.vim

```diff
@@ -0,0 +1,43 @@
+" Support script for the "formatter" feature
+" Maintainer:  Romain Lafourcade <romainlafourcade@gmail.com>
+" Last Change: 2025 Apr 18
+
+let s:keepcpo= &cpoptions
+set cpoptions&vim
+
+" Provide a default cross-platform function for :help 'formatexpr'
+function! format#FormatExpr()
+  if empty(&formatprg)
+    return 1
+  endif
+
+  let l1 = v:lnum
+  let l2 = (v:lnum + v:count) - 1
+
+  let tempfile = tempname()
+
+  call getline(l1, l2)->writefile(tempfile)
+
+  let output = systemlist(expandcmd(&formatprg) .. ' < ' .. tempfile)
```

Comment:
Why aren't you passing `getline(l1, l2)` directly to `systemlist()`? There's no need to write it to a temporary file:
```vim
silent let output = systemlist(expandcmd(&formatprg), getline(l1, l2))
```

This is explained under `:h :system()`.

Reply:
Because I didn't know about the optional `{input}` argument. Thank you.